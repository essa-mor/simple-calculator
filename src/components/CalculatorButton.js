import React from 'react';
import PropTypes from 'prop-types';

import disableFocus from './DisableFocus';

export function CalculatorButton({value , onClick, onFocus, className}){
	return <button type="button" value={value} onClick={onClick} onFocus={onFocus} className={className}>{value}</button>;
}

CalculatorButton.propTypes = {
	className: PropTypes.string , 
	value: PropTypes.string , 
	onClick: PropTypes.func,
	onFocus: PropTypes.func
};

export default disableFocus(CalculatorButton);