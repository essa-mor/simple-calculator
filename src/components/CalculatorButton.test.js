import React from 'react';
import renderer from 'react-test-renderer';
import { shallow  } from 'enzyme';

import { CalculatorButton } from './CalculatorButton';

describe('CalculatorButton test', () =>{
	it('Renders properly', () => {
		const tree = renderer.create(<CalculatorButton className="myclass" value="1" />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Renders properly', () => {
		const callback = jest.fn();
		const wrapper = shallow(<CalculatorButton onClick={callback} className="myclass" value="1" />);
		wrapper.find('button').simulate('click', 'event');
		expect(callback).toHaveBeenCalledWith('event');
	});
});