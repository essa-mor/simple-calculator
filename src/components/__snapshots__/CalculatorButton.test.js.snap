// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorButton test Renders properly 1`] = `
<button
  className="myclass"
  onClick={undefined}
  onFocus={undefined}
  type="button"
  value="1"
>
  1
</button>
`;
